xio-core/src/main/java/com/xjeffrose/xio/pipeline/XioBasePipeline.java

```diff
@@ -86,7 +87,10 @@ public void buildHandlers(
     } else {
       throw new RuntimeException("No codec configured");
     }
-    addHandler(pipeline, "distributed tracing", state.tracingHandler(appState));
+    val traceHandler = state.tracingHandler(appState);
+    if (traceHandler != null) {
+      addHandler(pipeline, "distributed tracing", traceHandler);
+    }
     addHandler(pipeline, "application codec", getApplicationCodec());
```

Comment:
`addHandler` does this if not null check, if you'd like to rename `addHandler` to something more descriptive, I'm on board.

Reply:
ok

Reply:
ok updated this to use addHandler for now (can't think of a better name)
also updated some of the other if checked handlers as well

Reply:
fixed